core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -1594,6 +1599,138 @@ private RecordCursorFactory createSpliceJoin(
+            return null;
+        }
+        // Ensure there's only one sort column
+        if (lastSeenOrderByModel.getOrderBy().size() != 1) {
+            return null;
+        }
+        // Ensure sort direction is ascending
+        IntList dirs = lastSeenOrderByModel.getOrderByDirection();
+        int dir = (dirs.size() > 0) ? dirs.getQuick(0) : ORDER_DIRECTION_ASCENDING;
+        if (dir != ORDER_DIRECTION_ASCENDING) {
+            return null;
+        }
+
+        // The ORDER BY node may be a column alias (LITERAL) - resolve it to the actual expression
+        ExpressionNode orderByNode = lastSeenOrderByModel.getOrderBy().getQuick(0);
+        ExpressionNode orderByExpr = orderByNode;
+        if (orderByNode.type == ExpressionNode.LITERAL) {
+            QueryColumn queryColumn = lastSeenOrderByModel.getAliasToColumnMap().get(orderByNode.token);
+            if (queryColumn != null) {
+                orderByExpr = queryColumn.getAst();
+            }
+        }
+        // Ensure that ORDER BY is an addition operation
+        if (orderByExpr.type != ExpressionNode.OPERATION || !"+".contentEquals(orderByExpr.token)) {
+            return null;
+        }
+        // The ORDER BY should be timestamp_column + offset_column, or vice versa
+        ExpressionNode lhs = orderByExpr.lhs;
+        ExpressionNode rhs = orderByExpr.rhs;
+        if (lhs == null || rhs == null) {
+            return null;
+        }
+
+        // Try to identify which operand is the timestamp from master and which is from slave
+        int timestampColumnIndex = -1;
+        ExpressionNode slaveColumnNode = null;
+
+        // Check if LHS is the designated timestamp column from master
+        if (lhs.type == ExpressionNode.LITERAL) {
+            int colIndex = masterMetadata.getColumnIndexQuiet(lhs.token);
+            if (colIndex >= 0 && colIndex == designatedTimestampIndex) {
+                timestampColumnIndex = colIndex;
+                slaveColumnNode = rhs;
+            }
+        }
+
+        // If not found, check if RHS is the designated timestamp column from master and LHS is from slave
+        if (timestampColumnIndex == -1 && rhs.type == ExpressionNode.LITERAL) {
+            int colIndex = masterMetadata.getColumnIndexQuiet(rhs.token);
+            if (colIndex >= 0 && colIndex == designatedTimestampIndex) {
+                timestampColumnIndex = colIndex;
+                slaveColumnNode = lhs;
+            }
+        }
+
+        if (slaveColumnNode == null || slaveColumnNode.type != ExpressionNode.LITERAL) {
+            return null;
+        }
+        int slaveColumnIndex = slaveMetadata.getColumnIndexQuiet(slaveColumnNode.token);
+        if (slaveColumnIndex == -1) {
+            return null; // Slave column not found
+        }
+        // Return the info needed to create the optimized factory.
+        // timestampColumnIndex will be -1 if no SELECT column matches the ORDER BY expression.
```

Comment:
The comment seems misleading: `timestampColumnIndex` will never be `-1` at this point. That's because it's initialized along with `slaveColumnNode` and there is the following check above:
```java
        if (slaveColumnNode == null || slaveColumnNode.type != ExpressionNode.LITERAL) {
            return null;
        }
```

Reply:
Removed comment in 3239a543455dbfd7b49537b4140bdedc1b630383